### PR TITLE
AUT-854: Set up SAML Proxy in Fargate

### PR DESCRIPTION
**Pull request by adityapahuja**
This change sets up a new service called SAML Proxy Fargate. This service will run in Fargate alongside with SAML Proxy running on EC2 instances.

Author: @adityapahuja

### CODE REVIEW

File: terraform/modules/hub/hub_saml_proxy.tf [hunk 1]
```diff
@@ -96,6 +108,50 @@ module "saml_proxy" {
...
+  tools_account_id  = var.tools_account_id
+  image_name        = "verify-saml-proxy"
+  certificate_arn   = var.wildcard_cert_arn
+  ecs_cluster_id    = aws_ecs_cluster.fargate-ecs-cluster.id
+  cpu               = 2048
+  # for a CPU of 2048 we need to set a RAM value between 4096 and 16384 (inclusive) that is a multiple of 1024.
+  memory  = ceil(max(var.saml_proxy_memory_hard_limit + 250, 4096) / 1024) * 1024
+  subnets = aws_subnet.internal.*.id
+  additional_task_security_group_ids = [
+    aws_security_group.scraped_by_prometheus.id,
```
Comment (philandstuff) [inline terraform/modules/hub/hub_saml_proxy.tf:149]:
```suggestion
```
This line is not needed.  The in-hindsight-badly-named `scraped_by_prometheus` security group opens port 9100 (node_exporter) and 9479 (beat-exporter), neither of which are relevant to fargate apps.

Comment (philandstuff) [reply terraform/modules/hub/hub_saml_proxy.tf:149]: (yes, we should remove it from config-v2 as well, I'll raise a PR for that)

Comment (philandstuff) [reply terraform/modules/hub/hub_saml_proxy.tf:149]: https://github.com/alphagov/verify-infrastructure/pull/483

File: terraform/modules/hub/files/prometheus/prometheus.yml [hunk 2]
```diff
@@ -73,6 +73,13 @@ scrape_configs:
     dns_sd_configs:
       - names:
           - 'frontend.hub.local'
+  - job_name: saml-engine
```
[the remark this answers is not in the document]
Comment (adityapahuja) [reply terraform/modules/hub/files/prometheus/prometheus.yml]: Done.

**Comment by philandstuff** [comment]
can we have a prometheus.yml change to add saml-proxy to the list? Similar to https://github.com/alphagov/verify-infrastructure/pull/468/files#diff-bcce5e55ca79022434c82a6a41c40820R111

**Comment by adityapahuja** [comment]
> can we have a prometheus.yml change to add saml-proxy to the list? Similar to https://github.com/alphagov/verify-infrastructure/pull/468/files#diff-bcce5e55ca79022434c82a6a41c40820R111

Yes. I still haven't finished updating the pull request. :)

**Comment by adityapahuja** [comment]
> can we have a prometheus.yml change to add saml-proxy to the list? Similar to https://github.com/alphagov/verify-infrastructure/pull/468/files#diff-bcce5e55ca79022434c82a6a41c40820R111

Fixed.

**Comment by philandstuff** [comment]
The new saml-proxy execution role needs access to parameter store and KMS.  See for example https://github.com/alphagov/verify-infrastructure/pull/468/commits/d783c059fca0dad0e3bea9e510cfa0180eeb5f87

**Comment by adityapahuja** [comment]
> The new saml-proxy execution role needs access to parameter store and KMS. See for example [d783c05](https://github.com/alphagov/verify-infrastructure/commit/d783c059fca0dad0e3bea9e510cfa0180eeb5f87)

Done

**Pull request merged**